pipelines/resources/deploy-ztp-edgeclusters-no-mirror.yaml

```diff
@@ -0,0 +1,291 @@
+    taskRef:
+      name: edgecluster-deploy-lso
+    params:
+      - name: edgeclusters-config
+        value: $(params.edgeclusters-config)
+      - name: kubeconfig
+        value: $(params.kubeconfig)
+      - name: ztp-container-image
+        value: $(params.ztp-container-image)
+      - name: mock
+        value: $(params.mock)
+      - name: pipeline-name
+        value: $(context.pipelineRun.name)
+    runAfter:
+      - deploy-workers
+    workspaces:
+      - name: ztp
+        workspace: ztp
+
+  # Deploy ODF
+  - name: deploy-odf
+    taskRef:
+      name: edgecluster-deploy-odf
+    params:
+      - name: edgeclusters-config
+        value: $(params.edgeclusters-config)
+      - name: kubeconfig
+        value: $(params.kubeconfig)
+      - name: ztp-container-image
+        value: $(params.ztp-container-image)
+      - name: mock
+        value: $(params.mock)
+      - name: pipeline-name
+        value: $(context.pipelineRun.name)
+    runAfter:
+      - deploy-lso
+    workspaces:
+      - name: ztp
+        workspace: ztp
+
+  # Deploy Disconnected Registry
+  - name: deploy-disconnected-registry
+    taskRef:
+      name: edgecluster-deploy-disconnected-registry-edgeclusters
+    params:
+      - name: edgeclusters-config
+        value: $(params.edgeclusters-config)
+      - name: kubeconfig
+        value: $(params.kubeconfig)
+      - name: ztp-container-image
+        value: $(params.ztp-container-image)
+      - name: mock
+        value: $(params.mock)
+      - name: pipeline-name
+        value: $(context.pipelineRun.name)
+    runAfter:
+      - deploy-odf
+    workspaces:
+      - name: ztp
+        workspace: ztp
+
+    # Deploy Disconnected Registry Pull secret
+  - name: deploy-disconnected-registry-ps
+    taskRef:
+      name: edgecluster-deploy-disconnected-registry-edgeclusters-ps
+    params:
+      - name: edgeclusters-config
+        value: $(params.edgeclusters-config)
+      - name: kubeconfig
+        value: $(params.kubeconfig)
+      - name: ztp-container-image
+        value: $(params.ztp-container-image)
+      - name: mock
+        value: $(params.mock)
+      - name: pipeline-name
+        value: $(context.pipelineRun.name)
+    runAfter:
+      - deploy-disconnected-registry
+    workspaces:
+      - name: ztp
+        workspace: ztp
+
+  # Deploy ICSP Edge-clusters Post
+  - name: deploy-icsp-edgeclusters-post
+    taskRef:
+      name: edgecluster-deploy-icsp-edgeclusters-post
+    params:
+      - name: edgeclusters-config
+        value: $(params.edgeclusters-config)
+      - name: kubeconfig
+        value: $(params.kubeconfig)
+      - name: ztp-container-image
+        value: $(params.ztp-container-image)
+      - name: mock
+        value: $(params.mock)
+      - name: pipeline-name
+        value: $(context.pipelineRun.name)
+    runAfter:
+      - deploy-disconnected-registry-ps
+    workspaces:
+      - name: ztp
+        workspace: ztp
+
+  # Deploy GPU Operator
+  - name: gpu-operator
```

Comment:
I'm still surprised that this task works. At this point, the cluster should point to its local registry, which doesn't have mirrored images. I don't see how it can find the `GPU` images to install the operator

Reply:
`[gpu-operator : gpu-operator] >>>> Grabbing info from Edge-clusters File   
[gpu-operator : gpu-operator] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>                                 
[gpu-operator : gpu-operator] Creating /workspace/ztp/deploy-ztp-edgeclusters-no-mirror-run-vcvnl/build/deploy-ztp-edgeclusters-no-mirror-run-vcvnl-gpu-operator-zc8bl/edgeclusters.yaml from EDGECLUSTERS_CONFIG 
[gpu-operator : gpu-operator] Kubeconfig file doesn't exist: creating one from token                                                                                                                              
[gpu-operator : gpu-operator] W1024 21:58:33.271621      47 loader.go:221] Config not found: /workspace/ztp/deploy-ztp-edgeclusters-no-mirror-run-vcvnl/build/deploy-ztp-edgeclusters-no-mirror-run-vcvnl-gpu-opera
tor-zc8bl/kubeconfig-hub                        
[gpu-operator : gpu-operator] W1024 21:58:33.271749      47 loader.go:221] Config not found: /workspace/ztp/deploy-ztp-edgeclusters-no-mirror-run-vcvnl/build/deploy-ztp-edgeclusters-no-mirror-run-vcvnl-gpu-opera
tor-zc8bl/kubeconfig-hub                                
[gpu-operator : gpu-operator] User "edgeclusters-deployer" set.
[gpu-operator : gpu-operator] Pull secret file /workspace/ztp/deploy-ztp-edgeclusters-no-mirror-run-vcvnl/build/deploy-ztp-edgeclusters-no-mirror-run-vcvnl-gpu-operator-zc8bl/pull-secret.json does not exist, gra
bbing from OpenShift                                                                                                                                                                                              
[gpu-operator : gpu-operator] # kubeconfig
[gpu-operator : gpu-operator] # raw-kubeconfig  
[gpu-operator : gpu-operator] >>>> Grabbing info from Edge-clusters File     
[gpu-operator : gpu-operator] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[gpu-operator : gpu-operator] Creating /workspace/ztp/deploy-ztp-edgeclusters-no-mirror-run-vcvnl/build/deploy-ztp-edgeclusters-no-mirror-run-vcvnl-gpu-operator-zc8bl/edgeclusters.yaml from EDGECLUSTERS_CONFIG
[gpu-operator : gpu-operator] Pull secret file /workspace/ztp/deploy-ztp-edgeclusters-no-mirror-run-vcvnl/build/deploy-ztp-edgeclusters-no-mirror-run-vcvnl-gpu-operator-zc8bl/pull-secret.json does not exist, gra
bbing from OpenShift                                                                                                                                                                                              
[gpu-operator : gpu-operator] Extract Kubeconfig for spk-factory-0
[gpu-operator : gpu-operator] # kubeconfig      
[gpu-operator : gpu-operator] # raw-kubeconfig                               
[gpu-operator : gpu-operator] >>>> Check if edgecluster spk-factory-0 require gpu-operator.
[gpu-operator : gpu-operator] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[gpu-operator : gpu-operator] Error: Cannot get keys of !!null, keys only works for maps and arrays
[gpu-operator : gpu-operator] >>>> Edgecluster spk-factory-0 does not require gpu-operator.                                                                                                                       
[gpu-operator : gpu-operator] >>>> This step is not neccesary, everything looks ready                    
[gpu-operator : gpu-operator] >>>>EOF                                  
[gpu-operator : gpu-operator] >>>>>>>                        
[gpu-operator : gpu-operator] >> GPU Operator Deployment done in: spk-factory-0
`

Reply:
Did you check the operator health? Pods all running? I'm quite confused by this.

If the operator is installed and working then I wonder if we are missing some ICSP for this catalogs :thinking: .

Reply:
There are no pods matching "gpu":  `oc get pod -A|grep -i gpu`
The message I pasted above says: Edgecluster spk-factory-0 does not require gpu-operator.



Also look at the subscriptions/csv:


`(.venv) [kni@provisionhost-0-0 ocs-ci]$ oc get subscriptions -A
NAMESPACE                 NAME                                                                      PACKAGE                       SOURCE          CHANNEL
metallb                   metallb-operator                                                          metallb-operator              ztpfw-catalog   stable
openshift-local-storage   local-storage-operator                                                    local-storage-operator        ztpfw-catalog   stable
openshift-nmstate         kubernetes-nmstate-operator                                               kubernetes-nmstate-operator   ztpfw-catalog   stable
openshift-storage         mcg-operator-stable-4.10-ztpfw-catalog-openshift-marketplace              mcg-operator                  ztpfw-catalog   stable-4.10
openshift-storage         ocs-operator-stable-4.10-ztpfw-catalog-openshift-marketplace              ocs-operator                  ztpfw-catalog   stable-4.10
openshift-storage         odf-csi-addons-operator-stable-4.10-ztpfw-catalog-openshift-marketplace   odf-csi-addons-operator       ztpfw-catalog   stable-4.10
openshift-storage         odf-operator                                                              odf-operator                  ztpfw-catalog   stable-4.10
ztpfw-registry            quay-operator                                                             quay-operator                 ztpfw-catalog   stable-3.7
(.venv) [kni@provisionhost-0-0 ocs-ci]$ oc get csv
No resources found in sasha namespace.
(.venv) [kni@provisionhost-0-0 ocs-ci]$ oc get csv -A
NAMESPACE                              NAME                                              DISPLAY                       VERSION               REPLACES                          PHASE
metallb                                metallb-operator.4.10.0-202210061437              MetalLB Operator              4.10.0-202210061437                                     Succeeded
openshift-local-storage                local-storage-operator.4.10.0-202210061437        Local Storage                 4.10.0-202210061437                                     Succeeded
openshift-nmstate                      kubernetes-nmstate-operator.4.10.0-202210061437   Kubernetes NMState Operator   4.10.0-202210061437                                     Succeeded
openshift-operator-lifecycle-manager   packageserver                                     Package Server                0.19.0                                                  Succeeded
openshift-storage                      mcg-operator.v4.10.7                              NooBaa Operator               4.10.7                mcg-operator.v4.10.6              Succeeded
openshift-storage                      ocs-operator.v4.10.7                              OpenShift Container Storage   4.10.7                ocs-operator.v4.10.6              Succeeded
openshift-storage                      odf-csi-addons-operator.v4.10.7                   CSI Addons                    4.10.7                odf-csi-addons-operator.v4.10.6   Succeeded
openshift-storage                      odf-operator.v4.10.7                              OpenShift Data Foundation     4.10.7                odf-operator.v4.10.6              Succeeded
ztpfw-registry                         quay-operator.v3.7.9                              Red Hat Quay                  3.7.9                 quay-operator.v3.7.8              Succeeded
`

Reply:
I don't have the proper line in my config: https://github.com/rh-ecosystem-edge/ztp-pipeline-relocatable/blob/main/contrib/gpu-operator/verify.sh#L26

Reply:
So you are not installing the GPU operator, which explains how this is working for you :) 

We may need to think about this pipeline a bit more because as it is, it will leave the cluster in a state where nothing can be installed because it's disconnected and pointing to a registry without images. Hence the suggestion of calling it a connected cluster and skip the `-post` task.